src/coreclr/tools/aot/ILCompiler.ReadyToRun/Compiler/ReadyToRunCompilationModuleGroupBase.cs

```diff
@@ -361,6 +359,9 @@ public sealed override bool GeneratesPInvoke(MethodDesc method)
             if (((EcmaMethod)method).Module.Equals(method.Context.SystemModule))
                 return true;
 
+            if (_versionBubbleModuleSet.Contains(method.Context.SystemModule))
+                return !Marshaller.IsMarshallingNotSupported(method);
```

Comment:
I believe we have an existing mechanism that rejects unsupported IL stubs already. Is this one redundant?

Reply:
Hi @gbalykov, Based on Jan's comments the only change required here is to check whether s.p.corelib.dll is part of the bubble. I will include that in https://github.com/dotnet/runtime/pull/55032, so we could validate with crossgen2 composite tests.

Reply:
@mangod9 I had some problems when `GeneratesPInvoke` returned true for methods which used `NotSupportedMarshaller`. This resulted in them not being compiled into ni.dll at all for some reason. I'll check with #55032

Reply:
So enabling stub generation for broader scenarios uncovered other issues. So plan is to merge the fix for BStr issue, and remove other Marshallers not implemented for R2R as a separate change.

Reply:
@mangod9 Ok, I'll update my PR when other marshallers are removed. This marshallers removal will be included in 6.0, right? Do you have milestone for correct marshallers inplementation for crossgen2? Will it be 6.0 too or 7.0?

Reply:
yeah hoping removal will be done in the next week. Correct implementation for cg2 most likely will be in 7

Reply:
@gbalykov the UTF8 and Unicode marshallers have been fixed. We will evaluate how to properly generate stubs for these cases in 7

Reply:
Thanks, I'll update this PR then